modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/entitlement/EntitlementSecurityTestCase.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.integration.test.entitlement;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URLEncoder;
+
+import junit.framework.Assert;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.commons.lang.StringEscapeUtils;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.identity.integration.common.utils.ISIntegrationTest;
+
+import javax.servlet.http.HttpServletResponse;
+
+public class EntitlementSecurityTestCase extends ISIntegrationTest {
+    private HttpClient httpClient;
+    private String value;
+    private String url;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+        super.init();
+        httpClient = HttpClientBuilder.create().build();
+        value = "<script>alert(1);</script>";
+        String encodedValue = URLEncoder.encode(value, "UTF-8");
+        String temp = backendURL.replaceAll("services/","carbon/policyeditor/prettyPrinter_ajaxprocessor.jsp?xmlString=");
+        url = temp + encodedValue;
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void atEnd() throws Exception {
+        httpClient = null;
+    }
+
+    @Test(alwaysRun = true, description = "Test whether cross site scripting (XSS) vulnerability is" + "resolved")
```

Comment:
Noted with thanks

On Fri, Aug 26, 2016 at 11:31 AM, Chamila Wijayarathna <
notifications@github.com> wrote:

> In modules/integration/tests-integration/tests-backend/src/
> test/java/org/wso2/identity/integration/test/entitlement/
> EntitlementSecurityTestCase.java
> <https://github.com/wso2/product-is/pull/680#discussion_r76367768>:
>
> > +    @BeforeClass(alwaysRun = true)
> > +    public void testInit() throws Exception {
> > +        super.init();
> > +        httpClient = HttpClientBuilder.create().build();
> > +        value = "<script>alert(1);</script>";
> > +        String encodedValue = URLEncoder.encode(value, "UTF-8");
> > +        String temp = backendURL.replaceAll("services/","carbon/policyeditor/prettyPrinter_ajaxprocessor.jsp?xmlString=");
> > +        url = temp + encodedValue;
> > +    }
> > +
> > +    @AfterClass(alwaysRun = true)
> > +    public void atEnd() throws Exception {
> > +        httpClient = null;
> > +    }
> > +
> > +    @Test(alwaysRun = true, description = "Test whether cross site scripting (XSS) vulnerability is" + "resolved")
>
> omit the + in description, better to improve the description as well.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wso2/product-is/pull/680/files/9c6bce33224fb812937a587439936d1f86d4620b#r76367768>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARvsJwCJfSy1Awg-CP6opOmflwJCpfmvks5qjoFOgaJpZM4Jtt3K>
> .
>



-- 
*Charini Vimansha Nanayakkara*
Software Engineer at WSO2

Mobile: 0714126293
E-mail: charinin@wso2.com
Blog: http://www.charini.me/

<http://wso2.com/signature>